### PR TITLE
Add more predicate operator to mbgl filter translations

### PR DESCRIPTION
**Pull request by boundsj**
Fixes https://github.com/mapbox/mapbox-gl-native/issues/6244

cc @1ec5 @frederoni

### CODE REVIEW

File: platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm [hunk 1]
```diff
@@ -9,16 +9,34 @@ @implementation NSComparisonPredicate (MGLAdditions)
...
+            if (self.rightExpression.constantValue)
+            {
+                auto filter = mbgl::style::NotEqualsFilter();
+                filter.key = self.leftExpression.keyPath.UTF8String;
+                filter.value = self.rightExpression.mgl_filterValue;
+                return filter;
+            }
+            else
+            {
+                auto filter = mbgl::style::NotHasFilter();
```
Comment (1ec5) [inline platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm]: Shouldn’t `%K != nil` be an `mbgl::style::HasFilter`?

File: platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm [hunk 2]
```diff
@@ -9,16 +9,34 @@ @implementation NSComparisonPredicate (MGLAdditions)
...
+            if (self.rightExpression.constantValue)
+            {
+                auto filter = mbgl::style::EqualsFilter();
+                filter.key = self.leftExpression.keyPath.UTF8String;
+                filter.value = self.rightExpression.mgl_filterValue;
+                return filter;
+            }
+            else
+            {
+                auto filter = mbgl::style::HasFilter();
```
Comment (1ec5) [inline platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm]: Shouldn’t `%K == nil` be an `mbgl::style::NotHasFilter`?

File: platform/darwin/src/NSPredicate+MGLAdditions.mm [hunk 3]
```diff
@@ -76,15 +76,11 @@
     }
     
     NSPredicate* operator()(mbgl::style::HasFilter filter) {
-        [NSException raise:@"Unsupported filter type"
-                    format:@"Cannot convert mbgl::style::HasFilter to NSPredicate"];
-        return nil;
+        return [NSPredicate predicateWithFormat:@"%K == nil", @(filter.key.c_str())];
```
Comment (1ec5) [inline platform/darwin/src/NSPredicate+MGLAdditions.mm]: Shouldn’t this be `%K != nil`?

File: platform/darwin/src/NSPredicate+MGLAdditions.mm [hunk 4]
```diff
@@ -76,15 +76,11 @@
...
     }
     
     NSPredicate* operator()(mbgl::style::NotHasFilter filter) {
-        [NSException raise:@"Unsupported filter type"
-                    format:@"Cannot convert mbgl::style::NotHasFilter to NSPredicate"];
-        return nil;
+        return [NSPredicate predicateWithFormat:@"%K != nil", @(filter.key.c_str())];
```
Comment (1ec5) [inline platform/darwin/src/NSPredicate+MGLAdditions.mm]: Shouldn’t this be `%K == nil`?

File: platform/darwin/test/MGLFilterTests.mm [hunk 5]
```diff
@@ -67,6 +74,51 @@ - (void)testAllPredicates
...
+    layer.predicate = containsPredicate;
+    XCTAssertEqualObjects(layer.predicate, expectedPredicate);
+    [self.mapView.style addLayer:layer];
+}
+
+- (void)testBetweenPredicate
+{
+    // core does not have a "contains" filter but we can achieve the equivilant by creating a set of greater than or equal / less than or equal
+    // filters for the lower and upper bounds (inclusive)
+    NSPredicate *expectedPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[[NSPredicate predicateWithFormat:@"%K >= 2", @"stroke-width"],
```
Comment (1ec5) [inline platform/darwin/test/MGLFilterTests.mm]: This could be expressed more succinctly as `%K >= 2 AND %K <= 3`

**Comment by boundsj** [comment]
> Pretty sure == nil and != nil are flipped, but looks good otherwise.

oops, will fix

**Pull request merged**